### ISSUE TITLE
Stuck at "Loading" plus more

### ISSUE THREAD

**Issue by jesjhoward** [jackmeyer/MMM-SleeperNFL#2]
Looks like I'm having a similar issue to gnwoodward in the other closed issue, but it might be a little different:

If I already have MM running and add the config, the module just stays at LOADING, but if I try to restart MM with your module in my config, the program just refuses to even start. Any ideas?

P.S. No pressure, but this would go a long way to convincing my wife to let me put a TV behind the mirror in my bathroom lolol

**Comment by jackmeyer** [comment]
@jesjhoward difficult to know without being able to see the logs, how are you running the MM software? Also, could you send over your config for the module?

If via PM2, 
Check your PM2 logs with `pm2 logs mm` or check the log files in their locations:

/home/pi/.pm2/pm2.log
/home/pi/.pm2/logs/mm-error-0.log
/home/pi/.pm2/logs/mm-out-0.log

**Comment by jesjhoward** [comment]
Thanks for the quick reply! I'm actually just in the testing phase for MM, so I'm running this in a docker container on a Linux VM in serveronly mode. I have several other modules running atm, but this one gives me a bit of trouble. I'll send the docker terminal output if that will help. It seems to be complaining about another module called "node-fetch", but I'm not sure if that's actually related or not

https://pastebin.com/5wDgBns8

I've boiled the config down to as small as I can get it, but almost any version of it I use, the container refuses to load

{
	module: "MMM-SleeperNFL",
	position: "bottom_left",
	config: {
		userId: "610626598424629248", //required
		leagueId: "1001675074686029824", //required
		}
}

**Comment by jackmeyer** [comment]
ah that's definitely related, navigate to the modules/MMM-SleeperNFL directory and run `npm install`

that _should_ be the only issue, let me know if that fixes it

**Comment by jesjhoward** [comment]
Unfortunately I get the same behavior. Should I see a directory called "node-fetch" running around somewhere?

**Comment by jackmeyer** [comment]
no, i'm basically referring to step 2 in the [Installation](https://github.com/jackmeyer/MMM-SleeperNFL#installation)

alternatively, you could try running `npm i node-fetch -g` to install the package globally.

**Comment by jesjhoward** [comment]
So I tried that with the same results. I also tried another docker container that runs MM and now I'm getting another message. I'm starting to think that it may have something to do with the container itself. I'll try it on an rpi tomorrow and see what that does. Thanks so much for the help!

**Comment by jesjhoward** [comment]
I know it's been a while, but I've finally returned to this project. I did try this on a pi4 with the same results. MM complains about not finding node-fetch, even after I have installed it manually. The installation path is /usr/local/lib/node-modules/node-fetch, but I'm not sure if that's where the module is looking for it or not

**Comment by jhartung2** [comment]
I was having the same issue @jesjhoward . Use `npm install node-fetch` in your MMM-SleeperNFL dir. 

However, this brings up another issue. 

> 
[ERROR] Error getting NFL state TypeError: fetch is not a function 
[2025-05-01 17:03:08.050] [ERROR] Error getting NFL scores TypeError: fetch is not a function 
[2025-05-01 17:03:08.565] [ERROR] Error getting rosterId TypeError: fetch is not a function 
[2025-05-01 17:03:08.566] [ERROR] Error getting league details TypeError: fetch is not a function 
[2025-05-01 17:03:08.568] [ERROR] Error getting matchup details TypeError: fetch is not a function 

Haven't dove into the code yet but may be a syntax issue. If someone knows off the top of their head that'd be nice otherwise I'll followup when I figure it out.

**Comment by jesjhoward** [comment]
I'll give it a shot. Let us know if you find anything. Thanks!

**Comment by jhartung2** [comment]
Okay, figured it out. @jackmeyer can correct me if I'm wrong, but the syntax for the node-fetch is for v2.X and if you run `npm install node-fetch` you will get the most up to date version (3.X). So I worked around that with  `cd Magicmirror/modules` and then grab v2 with `npm install node-fetch@2`

Right now I am getting an error for not showing starters, but I think this is likely having to do with it is technically offseason. When week 1 matchup is made I can always check back and comb through if anything seems wrong. 

Hope that helps.
